lightning-invoice/src/utils.rs

```diff
@@ -218,6 +215,81 @@ where
 	}
 }
 
+/// Utility to filter channels for an invoice, and return the corresponding route hints to include
+/// in the invoice.
+///
+/// The filtering ensures that the RouteHints returned only include the highest inbound capacity
+/// channel for the pubkey returned by the map_channels_to_pubkey. Channels with a lower inbound
+/// capacity than the invoice amount aren't included in the hints.
+///
+/// `channels`: The channels to filter.
+/// `min_inbound_capacity_required`: Defines the lowest inbound capacity a channel must have to not
+/// be filtered out.
+/// `empty_result_if_private_channels_exists`: If set to `true`, this entire function will return
+/// zero route hints if any of the inputted channels is a private channel.
+/// `channel_pubkey_selector`: Function that is called for every channel evaluated in the filtering
+/// that the required fields and has a high enough inbound_capacity. The route hint for the
+/// highest inbound capacity channel will be returned for each unique `PublicKey` returned by
+/// `channel_pubkey_selector`. The `ChannelDetails` parameter in `channel_pubkey_selector`, is the
+/// current channel evaluated for filtering.
+/// `route_hint_from_channel`: Mapping function that defines how a channel should be mapped into a
+/// `RouteHint`. The `ChannelDetails` parameter in `channel_pubkey_selector`, is the current channel evaluated
+/// for filtering. The `u64` parameter is the channel's `short_channel_id`.
+fn filter_channels<F1, F2>(
+	channels: Vec<ChannelDetails>,
+	min_inbound_capacity_required: Option<u64>,
+	empty_result_if_private_channels_exists: bool,
+	channel_pubkey_selector: F1,
```

Comment:
Given this always returns a constant (ie cause we call `filter_channels` inside the per-node loop in `create_phantom_invoice`), do we need this at all?

Reply:
In the `create_invoice_from_channelmanager` it actaully will vary, given that the channel counterparty node is different. However if my new understanding explained below is correct, it will always be the channel counterparty node id we are interested in, and this parameter wouldn't be needed at all! 
That would make the code a lot cleaner :)